src/SIM/SIMoutput.C

```diff
@@ -787,7 +798,7 @@ bool SIMoutput::writeGlvS2 (const Vector& psol, int iStep, int& nBlock,
                             idBlock++,iStep)) return false;
 
   if (haveAsol)
-    for (i = 0; i < nf && j < sMAX && !sID[j].empty(); i++, j++)
+    for (i = 0; i < sAna && j < sMAX && !sID[j].empty(); i++, j++)
```

Comment:
And this can then be `for (i = 0; j < sID.size() && !sID[j].empty(); i++; j++)`

Reply:
attempt to fix done. need to check that it works and i do not have the code here today.